runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -1802,8 +1798,17 @@ TR_VectorAPIExpansion::transformIL(bool checkBoxing)
          int32_t elementSize = OMR::DataType::getSize(elementType);
          numLanes = bitsLength/8/elementSize;
 
-         // TODO: add an assert that unboxing is not needed in this case since the temp
-         // would not be vectorized
+         if (boxingAllowed())
+            {
+            TR::DataType elementTypeTmp;
+            int32_t bitsLengthTmp;
+            vapiObjType objectTypeTmp;
+            bool scalarizedTmp;
```

Comment:
you don't need to initialize these 4 Tmp(s)? and, they are not used after coming back from the call.  we experienced a very elusive bug when a call is made with arguments not initialized (although the code was correct): OpenXL optimizer just generated garbage code.

Reply:
These are values set by the call and they are not used in this case. Not sure how to make that nicer.

Reply:
We do it in other places too. In the future we can return a structure filled in with those values.

Reply:
For now, I gave them *Tmp names to make it more obvious.